net-reporter-api/src/api/http_responses/http_responses_filters.rs

```diff
@@ -0,0 +1,402 @@
+}
+
+impl Typed for HttpResponsesFiltersDTO {
+    fn get_data_type() -> &'static str {
+        DATA_TYPE
+    }
+
+    fn get_type(&self) -> &str {
+        Self::get_data_type()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use ion_rs::element::reader::ElementReader;
+    use ion_rs::types::IntAccess;
+    use ion_rs::IonType;
+    use ion_rs::IonReader;
+    use ion_rs::ReaderBuilder;
+    use ion_rs::StreamItem;
+    
+    use net_core_api::encoder_api::Encoder;
+    use net_core_api::decoder_api::Decoder;
+
+    use crate::api::http_responses::http_responses_filters::HttpResponsesFiltersDTO;
+
+    
+    #[test]
+    fn reader_correctly_read_encoded_http_responses_filters_0() {
+        let endpoints = vec!["0.0.0.0".to_string(), "1.1.1.1".to_string()];
+        const INCLUDE_ENDPOINTS_MODE: bool = true;
+        let bytes_lower_bound = Some(100);
+        
+        let filters = HttpResponsesFiltersDTO::new(
+            &[],
+            None,
+            &endpoints,
+            Some(INCLUDE_ENDPOINTS_MODE),
+            bytes_lower_bound,
+            None,
+        );
+        
+        let mut binary_user_reader = ReaderBuilder::new().build(filters.encode()).unwrap();
+
+        assert_eq!(StreamItem::Value(IonType::Struct), binary_user_reader.next().unwrap());
+        binary_user_reader.step_in().unwrap();
+        
+        assert_eq!(StreamItem::Null(IonType::Bool), binary_user_reader.next().unwrap());
+        assert_eq!("include_http_responses_mode", binary_user_reader.field_name().unwrap());
+
+        assert_eq!(StreamItem::Value(IonType::Bool), binary_user_reader.next().unwrap());
+        assert_eq!("include_endpoints_mode", binary_user_reader.field_name().unwrap());
+        assert_eq!(INCLUDE_ENDPOINTS_MODE,  binary_user_reader.read_bool().unwrap());
+
+        assert_eq!(StreamItem::Value(IonType::List), binary_user_reader.next().unwrap());
+        assert_eq!("endpoints", binary_user_reader.field_name().unwrap());
+        binary_user_reader.step_in().unwrap();
+        let elements = binary_user_reader.read_all_elements().unwrap();
+        assert_eq!(elements.len(), endpoints.len());
+        for (element, core_endpoint) in elements.iter().zip(endpoints.as_slice()) {
+            let endpoint = element.as_string().unwrap();
+            assert_eq!(endpoint, *core_endpoint);
+        }
+        binary_user_reader.step_out().unwrap();
+
+        assert_eq!(StreamItem::Value(IonType::Int), binary_user_reader.next().unwrap());
+        assert_eq!("bytes_lower_bound", binary_user_reader.field_name().unwrap());
+        assert_eq!(*bytes_lower_bound.as_ref().unwrap(),  binary_user_reader.read_i64().unwrap());
+
+        assert_eq!(StreamItem::Null(IonType::Int), binary_user_reader.next().unwrap());
+        assert_eq!("bytes_upper_bound", binary_user_reader.field_name().unwrap());
+
+        binary_user_reader.step_out().unwrap();
+    }
+
+    
+    #[test]
+    fn reader_correctly_read_encoded_http_responses_filters_1() {
```

Comment:
Numbers :cry: 
Not gonna lie, hate numbers like this in names. No actual meaning.